website/Contribution/CONTRIBUTING.rst

```diff
@@ -1066,6 +1066,93 @@ test cases.
 you should name it ``module_name_example`` (ie: `cms_form` and `cms_form_example`).
 In this way coverage analysis will ignore this extra module by default.
 
+Avoid Flaky Tests
+=================
+
+Flaky tests are unit tests that produce inconsistent or unreliable results.
+These tests may pass or fail intermittently, even when the code being tested
+hasn't changed. Flakiness can be caused by various factors, such as race
+conditions, external dependencies, or non-deterministic behavior in the code.
+How to avoid them?
+
+You can find online some general studies and guidelines about the subject:
+
+- https://docs.gitlab.com/ee/development/testing_guide/flaky_tests.html
+- https://semaphoreci.com/community/tutorials/how-to-deal-with-and-eliminate-flaky-tests
+- https://martinfowler.com/articles/nonDeterminism.html
+
+In the context of Odoo, the following are some tips to avoid flaky tests:
+
+Beware with ``subTest`` pollution
+---------------------------------
+
+If you test uses ``subTest``, keep in mind that the DB transaction and the
+environmental cache aren't reset between subtests.
+
+Remember that some decorators are internally implemented using subtests.
+
+A bad example:
+
+.. code-block:: python
+
+  from odoo.tests.common import TransactionCase, users
+
+  class TestSomething(TransactionCase):
+    @users('admin', 'demo')
+    def test_something(self):
+      """2nd run will fail with duplicate key on login field."""
+      self.env["res.users"].create({"name": "Foo", "login": "foo"})
+
+Avoid dynamic dates
+-------------------
+
+If you are testing a feature that depends on a date, you should avoid using
+``datetime.now()`` or ``datetime.today()``. Instead, use fixed dates. A way to
+do this is to use `the freezegun library <https://pypi.org/project/freezegun/>__`:
```

Comment:
Maybe it's worth it to mention this library is already in Odoo's requirements, so a standard installation won't require installing any additional library.